source/common/filesystem/watcher_impl.cc

```diff
@@ -1,93 +1,7 @@
-#include "common/filesystem/watcher_impl.h"
+// NOLINT(namespace-envoy)
+#if defined(__linux__)
+#include "common/filesystem/watcher_impl_linux.cc"
```

Comment:
Can we use select based Bazel target srcs/hdrs to avoid this kind of compile time conditioning?